dvc/cli.py

```diff
@@ -116,11 +151,96 @@ def parse_args(self, args=None, namespace=None):
+
+        Returns:
+            A dictionary with public dvc commands mapped to a list containing
+            dvc subcommands (if they exist).
+
+            A list containing hidden dvc commands to be ignored when finding
+            suggestions.
+        """
+        cmd_choices = {}
+        hidden_cmds = []
+
+        parser_actions = self._actions  # pylint: disable=protected-access
+        for parser_action in parser_actions:
+            if parser_action.dest == "help":
+                # treat -h, --help as command choices
+                for option in parser_action.option_strings:
+                    cmd_choices[option] = []
+            elif parser_action.dest == "cmd":
+                for cmd, subparser in parser_action.choices.items():
+                    if not subparser.add_help:
+                        hidden_cmds.append(cmd)
+                        continue
+                    cmd_choices[cmd] = []
+                    actions = (
+                        subparser._actions  # pylint: disable=protected-access
+                    )
+                    for action in actions:
+                        if not isinstance(action.choices, dict):
+                            # NOTE: we are only interested in subcommands
+                            continue
+                        cmd_choices[cmd].extend(action.choices.keys())
+
+        return cmd_choices, hidden_cmds
+
+    def _find_cmd_suggestions(self, args):
+        """Find similar command suggestions for commands and subcommands
+        that contain typos. Show suggestions for public commands, do not show
+        suggestions for hidden commands.
+
+        Args:
+            args: argument strings.
+
+        Raises:
+            dvc.exceptions.DvcParserError: raised for found suggestions.
+        """
+        cmd_choices, hidden_cmds = self._get_cmd_choices()
+
+        # NOTE: Check top level dvc commands for suggestions
+        # E.g. `dvc commti` would display
+        # The most similar command is
+        #         commit
+        if (
+            len(args) >= 1
+            and args[0] not in cmd_choices
+            and args[0] not in hidden_cmds
+        ):
+            cmd_suggestion_msg = _get_cmd_suggestion_msg(
+                args[0], list(cmd_choices.keys())
+            )
+            logger.error(cmd_suggestion_msg)
+            raise DvcParserError
+
+        # NOTE: Check nested dvc subcommands for suggestions
+        # E.g. `dvc remote modfiy` would display
+        # The most similar command is
+        #         remote modify
+        if len(args) == 2 and args[0] in cmd_choices:
+            sub_cmd_choices = cmd_choices[args[0]]
+            if sub_cmd_choices and args[1] not in sub_cmd_choices:
+                cmd_suggestion_msg = _get_cmd_suggestion_msg(
+                    args[1], sub_cmd_choices, args[0]
+                )
+                logger.error(cmd_suggestion_msg)
+                raise DvcParserError
```

Comment:
This will fallthrough if len(args) is bigger than 2 and will not raise a error up the stack.